uPortal-portlets/src/main/java/org/apereo/portal/portlets/search/SearchPortletController.java

```diff
@@ -271,28 +312,26 @@ public void performSearch(
 
             //Too many searches in the last minute, fail the search
             if (searchCounterCache.size() > this.maximumSearchesPerMinute) {
-                logger.debug(
-                        "Rejecting search for '{}', exceeded max queries per minute for user",
-                        query);
-
-                if (!ajax) {
-                    response.setRenderParameter("hitMaxQueries", Boolean.TRUE.toString());
-                    response.setRenderParameter("query", query);
-                } else {
-                    // For Ajax return to a nonexistent file to generate the 404 error since it was easier for the
-                    // UI to have an error response.
-                    final String contextPath = request.getContextPath();
-                    response.sendRedirect(contextPath + AJAX_MAX_QUERIES_URL);
-                }
-                return;
+                tooManyQueries = true;
             }
         }
+        return tooManyQueries;
+    }
 
-        // construct a new search query object from the string query
-        final SearchRequest queryObj = new SearchRequest();
-        queryObj.setQueryId(queryId);
-        queryObj.setSearchTerms(query);
+    private Cache<String, Boolean> getSearchCounterCache(PortletSession session) {
+        Cache<String, Boolean> searchCounterCache;
+        synchronized (org.springframework.web.portlet.util.PortletUtils.getSessionMutex(session)) {
+            searchCounterCache = (Cache<String, Boolean>) session.getAttribute(SEARCH_COUNTER_NAME);
+            if (searchCounterCache == null) {
+                searchCounterCache =
+                        CacheBuilder.newBuilder().expireAfterAccess(1, TimeUnit.MINUTES).build();
```

Comment:
:question: should this be configurable?

Reply:
Probably but this PR is focused on adding a new API endpoint, not updating the old search controller.